File: Easy/387. First Unique Character in a String/firstUniqChar.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        //利用map統計每個字母出現的次數後再遍歷一次找出第一個只出現一次的字母並且回傳她的index
        unordered_map<int, int> m;
        for(char c : s){
            m[c]++;
        }
        for(int i = 0; i < s.size(); i++){
            if(m[s[i]] == 1){
                return i;
            }
        }
        return -1;
    }
};

File: Easy/387. First Unique Character in a String/firstUniqChar.java
class Solution {
    public int firstUniqChar(String s) {
        //題目說只會出現小寫所以用一個陣列存26個字母分別出現的次數
        //再重新遍歷找出第一個出現一次的字母 並回傳她的index
        //time complexity: O(n)
        //Space : O(1)
        int [] a = new int [26];
        for(char c : s.toCharArray()){
            a[c - 'a']++;
        }
        for(int i = 0; i < s.length(); i++){
            if(a[s.charAt(i) - 'a'] == 1){
                return i;
            }
        }
        return -1;
    }
}